### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by fadyanwar** [microsoft/Power-Fx#67] [Q&A] [answered]
Hi, I'm reading through the source code and it looks quite intricate with very little to no inline documentation, so I was wondering if Microsoft is going to release any documentation in this regard in the near future, for example to detail how to add a new built in function while complying with PowerFx used design patterns. Thanks.

Fady

**Comment by MikeStall** [comment, marked as answer]
We're working on it! If there are specific areas that you'd like to see more documentation or samples, let us know. 

The sample repl shows how to add a basic custom function: https://github.com/microsoft/power-fx-host-samples/blob/417a3100b96420167004e259c06ed8c4d81d7420/Samples/ConsoleREPL/ConsoleREPL.cs#L22

**Reply by fadyanwar** [reply]
Sounds good and thanks for the tip. I'm quite interested in the built in functions as in adding new ones or modifying the behavior of existing ones in the core nuget.
